### ISSUE TITLE
HTTP 429 Error.

### ISSUE THREAD

**Issue by EHMilon** [ai-genie/chatgpt-vscode#106]
### Verify it's not a duplicate bug report

- [X] I searched for similar bugs reported on GitHub and nobody has reported a similar issue. <a href="https://github.com/ai-genie/vscode-chatgpt/issues?q=" target="_blank">Search all issuees</a>.
- [X] I read <a href="https://github.com/ai-genie/vscode-chatgpt#troubleshooting" target="_blank">Troubleshooting guide on Readme</a> and there is not a guide on how to solve my problem.

### **When i search something this error will happen:**
HTTP 429 Too Many Requests. Too many requests try again later. You can clear your API Key from VS Code's [secrets storage](https://code.visualstudio.com/api/references/vscode-api#SecretStorage) with Genie: Clear API Key command. Potential reasons:

You exceeded your current quota, please check your plan and billing details
You are sending requests too quickly
The engine is currently overloaded, please try again later.
See https://platform.openai.com/docs/guides/error-codes for more details.
{
    "error": {
        "message": "You exceeded your current quota, please check your plan and billing details.",
        "type": "insufficient_quota",
        "param": null,
        "code": null
    }
}
 

### Please tell us if you have customized any of the extension settings or whether you are using the defaults.

default

### Additional context

i tried to reinstall, clear API still it don't work.
<img width="652" alt="image" src="https://github.com/ai-genie/chatgpt-vscode/assets/107425940/b5f55f8c-58aa-4c92-9527-fd55bf3983d2">

**Comment by Mkp4coder** [comment]
Hii Milon, I'm facing exactly the same issue. I'm trying again and again but I can't be able to solve this. Please text me back when you will succeed in solving this Bug.

**Comment by stinghxk** [comment]
Perhaps your free trial has expired if you are not a paid account.

**Comment by vogofox** [comment]
I have exactly the same issue.
(not because of trial expire, this should be something else...)

**Comment by EHMilon** [comment]
"I haven't received any support from the Genie AI developer. It has been two days and I still haven't heard back."

**Comment by EHMilon** [comment]
> Perhaps your free trial has expired if you are not a paid account.

are you sure. there are no announcement for paid usage genie ai extension.

**Comment by EHMilon** [comment]
i am uninstalling Genie AI extension.

**Comment by xxyy110** [comment]
I think it is your openai free account expired. I have exact the same error and I also tried different openai backed vs code extension, none worked.
The openai free account expires in 3 months. After that you need to setup paid account and use pay as you go plan.

**Comment by stinghxk** [comment]
> > Perhaps your free trial has expired if you are not a paid account.
> 
> are you sure. there are no announcement for paid usage genie ai extension.

Sorry for the confusion. I mean the openAI account you use. It has nothing to do with this extension.

**Comment by contracourse** [comment]
> 

I have the same problem but can still use chatgpt on the chat.openai website.

**Comment by EHMilon** [comment]
Now I understand.

**Comment by aleksandr-ideasoft** [comment]
Same problem, but API key is valid and quota actually not exceeded.

**Comment by bagrounds** [comment]
Same problem. Genie works fine on my work machine, but I got 429s on first attempt on my personal machine.
openai account created yesterday, definitely not expired.
I can chat with the chatbot on the website fine.
I tried creating 2 brand new API keys, same results with both.
My API Keys in the chat GPT settings interface (website) show the keys have never been used.
I wonder if copy/pasting the key is somehow not working properly... maybe some extra characters added or something... kinda hard to debug when the API key is obfuscated in the VSCode prompt.
